### PR TITLE
Extract a base component for blacklight

### PR DESCRIPTION
**Pull request by barmintor**
Backports #2679 to 7.x

### CODE REVIEW

**Comment by barmintor** [comment]
Since we've made a decision on `main` regarding base components, I've decided to take the more conservative approach to overrides and substitute in a `ViewComponent::Compiler` subclass for descendants of `Blacklight::Component` in this PR.

This allows simple overrides of component templates that exist in the engine by placing them in the same place in the installing application. If a file in the installing application doesn't have a same-named analog in the engine, it won't be recognized - this has an impact on translation sidecars if the installing app is more expansive than the engine in language scope. But the Compiler interface is more stable, and if `ViewComponent` commits to an interface for sidecar files we can revisit without disrupting existing apps.

File: app/components/blacklight/constraints_component.rb [hunk 1]
```diff
@@ -1,9 +1,7 @@
 # frozen_string_literal: true
 
 module Blacklight
-  class ConstraintsComponent < ::ViewComponent::Base
-    include Blacklight::ContentAreasShim
```
Comment (jcoyne) [inline app/components/blacklight/constraints_component.rb:5]: Was this exclusion intentional?

Comment (barmintor) [reply app/components/blacklight/constraints_component.rb:5]: Added a commit, will squash after review

File: lib/blacklight/component.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Blacklight
+  class Component < ViewComponent::Base
+    class << self
```
Comment (jcoyne) [inline lib/blacklight/component.rb:5]: Please add documentation around what this is doing and why.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: lib/blacklight/component.rb (hunk 2 above)
Comment (barmintor) [reply lib/blacklight/component.rb:5]: Documentary comments added.

**Pull request merged**